client/extensions/woocommerce/app/dashboard/widgets/stats-widget/index.js

```diff
@@ -0,0 +1,366 @@
+import { sortBy, find } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import { dashboardListLimit } from 'woocommerce/app/store-stats/constants';
+import DashboardWidget from 'woocommerce/components/dashboard-widget';
+import { getLink } from 'woocommerce/lib/nav-utils';
+import { getPreference } from 'state/preferences/selectors';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { getSiteStatsNormalizedData } from 'state/stats/lists/selectors';
+import { getQueries } from './queries';
+import {
+	getUnitPeriod,
+	getStartPeriod,
+	getDelta,
+	getDeltaFromData,
+	getEndPeriod,
+	getConversionRateData,
+} from 'woocommerce/app/store-stats/utils';
+import List from './list';
+import QueryPreferences from 'components/data/query-preferences';
+import QuerySiteStats from 'components/data/query-site-stats';
+import { savePreference } from 'state/preferences/actions';
+import SelectDropdown from 'components/select-dropdown';
+import Stat from './stat';
+
+class StatsWidget extends Component {
+	static propTypes = {
+		site: PropTypes.shape( {
+			name: PropTypes.string.isRequired,
+			slug: PropTypes.string.isRequired,
+		} ),
+		unit: PropTypes.string,
+		queries: PropTypes.object,
+		orderData: PropTypes.object,
+		referrerData: PropTypes.array,
+		topEarnersData: PropTypes.array,
+		visitorData: PropTypes.array,
+		productData: PropTypes.array,
+		saveDashboardUnit: PropTypes.func,
+	};
+
+	handleTimePeriodChange = option => {
+		const { saveDashboardUnit } = this.props;
+		saveDashboardUnit( option.value );
+	};
+
+	dateForDisplay = () => {
+		const { translate, unit, moment } = this.props;
+
+		const localizedDate = moment( moment().format( 'YYYY-MM-DD' ) );
+		let formattedDate;
+		switch ( unit ) {
+			case 'week':
+				formattedDate = translate( '%(startDate)s - %(endDate)s', {
+					context: 'Date range for which stats are being displayed',
+					args: {
+						// LL is a date localized by momentjs
+						startDate: localizedDate
+							.startOf( 'week' )
+							.add( 1, 'd' )
+							.format( 'LL' ),
+						endDate: localizedDate
+							.endOf( 'week' )
+							.add( 1, 'd' )
+							.format( 'LL' ),
+					},
+				} );
+				break;
+
+			case 'month':
+				formattedDate = localizedDate.format( 'MMMM YYYY' );
+				break;
+
+			case 'year':
+				formattedDate = localizedDate.format( 'YYYY' );
+				break;
+
+			default:
+				// LL is a date localized by momentjs
+				formattedDate = localizedDate.format( 'LL' );
+		}
+
+		return formattedDate;
+	};
+
+	renderTitle = () => {
+		const { site, translate, unit } = this.props;
+
+		const options = [
+			{ value: 'day', label: translate( 'day' ) },
+			{ value: 'week', label: translate( 'week' ) },
+			{ value: 'month', label: translate( 'month' ) },
+		];
+
+		const dateDisplay = this.dateForDisplay();
+
+		return (
+			<Fragment>
+				<span>
+					{ translate( '%(siteName)s in the last {{timePeriod/}}', {
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?